openslides_backend/services/authentication.py

```diff
@@ -23,24 +26,41 @@ def get_user(self, headers: Headers) -> int:
         self.logger.debug(
             f"Start request to authentication service with the following data: {headers}"
         )
-        response = requests.post(
-            self.url, data=json.dumps(headers.to_wsgi_list()), headers=self.headers
-        )
-        if not response.ok:
-            raise AuthenticationException(
-                f"Authentication service sends HTTP {response.status_code}."
-            )
-        try:
-            body = response.json()
-        except JSONDecodeError:
-            raise AuthenticationException(
-                "Bad response from authentication service. Body does not contain JSON."
-            )
-        self.logger.debug(f"Get repsonse: {body}")
+
+        if self.is_auth_accessible(self.url):
+            request_data = json.dumps(headers.to_wsgi_list())
+            response = requests.post(self.url, data=request_data, headers=self.headers)
+            if not response.ok:
+                raise AuthenticationException(
+                    f"Authentication service sends HTTP {response.status_code}."
+                )
+            try:
+                body = response.json()
+            except JSONDecodeError:
+                raise AuthenticationException(
+                    "Bad response from authentication service. Body does not contain JSON."
+                )
+            self.logger.debug(f"Get repsonse: {body}")
+            try:
+                user_id = body["user_id"]
+            except (TypeError, KeyError):
+                raise AuthenticationException(
+                    "Empty or bad response from authentication service."
+                )
+        else:
+            self.logger.debug(f"Auth cannot be reached. Fall back to guest mode.")
+            user_id = GUEST_USER_ID
+        return user_id
+
+    def is_auth_accessible(self, url: str) -> bool:
+        """
+        Checks if authentication service is accessible.
+        """
         try:
-            user_id = body["user_id"]
-        except (TypeError, KeyError):
-            raise AuthenticationException(
-                "Empty or bad response from authentication service."
+            response = requests.head(url)
```

Comment:
Are you sure you want to send two request to the auth service in the healthy situation?

You can also send the real request immediately and check if it works. There is no caching etc, so there is no need for an head request.

Also, your code does not work in all cases. If the auth service gets down after the head request is send, but before the real request is send, then you will get a AuthenticationException.

Reply:
> Are you sure you want to send two request to the auth service in the healthy situation?
> You can also send the real request immediately and check if it works. There is no caching etc, so there is no need for an head request.

I tried that, but did not find any meaningful exceptions in that case. 

> Also, your code does not work in all cases. If the auth service gets down after the head request is send, but before the real request is send, then you will get a AuthenticationException.

Sounds not too realistic.